math/normaliz/Portfile

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               github 1.0
+
+github.setup            Normaliz Normaliz 3.10.0
+name                    normaliz
```

Comment:
please add ```github.tarball_from ????```, ideally it would be "releases" if upstream provides it, otherwise "tarball" would be appropriate (applies to other ports as well). This will require to recalculate the `checksums` most likely

Reply:
@reneeotten On a side note, why not call that “tags”, as that is called in reality, but use a weird-sounding and carrying no coherent meaning “tarball from tarball”?

Reply:
I think I've linked to a post fromRyan describing the differences before, but cannot find that anymore now. You could read the github PG to check the exact details but in short:
* releases: a manually generated release tarball by upstream (preferred)
* tarball: an automatically generated tarball by GitHub for an tag
* tag: fetches from the actual tag, which is theoretically mutable if upstream force-pushes or whatever

Reply:
Valid values for `github.tarball_from` are `tarball` (the default, which should not be used) and its deprecated synonym `tags`, `archive` (which should be used if upstream does not provide a release or download distfile), `releases` (which should be used if upstream provides a release distfile), or `downloads` (which should be used for ancient projects whose distfiles are hosted on the [discontinued GitHub "uploads" service](https://github.blog/2012-12-12-goodbye-uploads/)).

The four values `tarball`, `archive`, `releases`, and `downloads` are so named because that word appears in the download URL (or did at the time the name was chosen; the `tarball` URL has since changed and no longer includes that word).

`tarball` and `archive` download automatically-generated tarballs from any git commit-ish, i.e. a commit hash, tag name, or branch name. Branch names should not be used since that would break as soon as upstream commits to that branch. You're right that using tag names would theoretically break if upstream were to recreate a tag; upstream should not do that but it has happened in a few ports. `release` downloads from a file the developer has attached to the GitHub release attached to a git tag. `downloads` downloads from a file the developer has uploaded to their project's uploads area back when GitHub allowed that to be done.

`tarball` should not be renamed back to `tags` both because it can download from more places than just tags (i.e. commits or branches) and because `archive` can also download from tags, commits, and branches.

I acknowledge that naming the portgroup option `tarball_from` was a bad idea, not because of the reason you state but because distfiles are not always tarballs. A redesigned github 2.0 portgroup could fix this and many other design mistakes but the use of the portgroup is already so pervasive that changing each port to use a redesigned portgroup would be a lot of work.